### PR TITLE
Staff of change can now transform people into ANY mob

### PR DESCRIPTION
**Pull request by Buggy123**
OHHHH BOY PLEASE DO NOT MERGE THIS UNTIL I GET A CHANCE TO TEST IT BECAUSE 9/10 THIS BREAKS SHIT

Exactly what it says on the tin, change projectiles (staff/wand of change, magicarp, etc) will now transform you into a completely random /mob. Probably broken as all fuck.

🆑 
tweak: The Wizard Federation has found a high-quality source for the key ingredient to polymorph magic: Changling hearts. Be advised that all polymorph magic has become substantially more powerful and unpredictable.
/🆑

### CODE REVIEW

**Comment by coiax** [comment]
Subtypes of /mob include observers.

**Comment by vuonojenmustaturska** [comment]
This is completely broken and you're just breaking the few transformations that work.

**Comment by Buggy123** [comment]
This will probably need a exceptions list yeah, feel free to suggest things.

**Comment by XDTM** [comment]
These ain't chems, there's a lot of prototype-only mobs or pseudomobs (like brainmobs)

**Comment by Buggy123** [comment]
I'll be adding any of those I can find to the list, yeah.

**Comment by XDTM** [comment]
It's easier to whitelist than blacklist here, you can do stuff like subtypesof (mob/living/simple_animal) if you want a full list

**Comment by Buggy123** [comment]
Hmm, fair enough. I'll do that instead.

**Comment by ExcessiveUseOfCobblestone** [comment]
> it may need an exception list

Gee why didn't anyone think of that!

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -114,156 +114,8 @@
...
-					if("lizard")
-						path = /mob/living/simple_animal/hostile/lizard
-					if("fox")
-						path = /mob/living/simple_animal/pet/fox
-					if("butterfly")
-						path = /mob/living/simple_animal/butterfly
-					if("cak")
-						path = /mob/living/simple_animal/pet/cat/cak
-					if("chick")
-						path = /mob/living/simple_animal/chick
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/projectile/magic.dm]: remind me to report whoever made this code, ill make it look better if this pr gets closed

File: code/modules/projectiles/projectile/magic.dm [hunk 2]
```diff
@@ -114,156 +114,8 @@
...
-			new_mob = new path(M.loc)
-
-		if("humanoid")
-			new_mob = new /mob/living/carbon/human(M.loc)
-
-			var/datum/preferences/A = new()	//Randomize appearance for the human
-			A.copy_to(new_mob, icon_updates=0)
-
-			var/mob/living/carbon/human/H = new_mob
-			if(prob(50))
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/magic.dm]: rip species code

File: code/modules/projectiles/projectile/magic.dm [hunk 3]
```diff
@@ -114,156 +114,8 @@
...
-					new_mob = new /mob/living/simple_animal/drone/polymorphed(M.loc)
-			if(issilicon(new_mob))
-				new_mob.gender = M.gender
-				new_mob.invisibility = 0
-				new_mob.job = "Cyborg"
-				var/mob/living/silicon/robot/Robot = new_mob
-				Robot.mmi.transfer_identity(M)	//Does not transfer key/client.
-				Robot.clear_inherent_laws(0)
-				Robot.clear_zeroth_law(0, 0)
-				Robot.connected_ai = null
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/magic.dm]: rip cyborg handling

File: code/modules/projectiles/projectile/magic.dm [hunk 4]
```diff
@@ -114,156 +114,8 @@
...
 
-	if(iscyborg(M))
-		var/mob/living/silicon/robot/Robot = M
-		if(Robot.mmi)
-			qdel(Robot.mmi)
-		Robot.notify_ai(NEW_BORG)
-	else
-		for(var/obj/item/W in contents)
-			if(!M.dropItemToGround(W))
-				qdel(W)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/magic.dm]: hope we didn't need this code

Comment (Buggy123) [reply code/modules/projectiles/projectile/magic.dm]: naaaaaah

**Comment by Buggy123** [comment]
ignor travis my code is flawless and perfect

**Comment by ExcessiveUseOfCobblestone** [comment]
What's the end goal here that can't be completed by adding something to the current list?

**Comment by XDTM** [comment]
You 're losing out on a lot of customization if you do this; just remove all the simple mobs from the current list and make a collective pick from simple animals removing the blacklisted ones, and keep the existing categories as they are.

**Comment by GunHog** [comment]
Ehh, I think you are better off just expanding the possible mobs approach. Being overtly inclusive means that you will be flooded with undesirables. It is probably a pain to keep having to add more mobs to the staff when new mobs are added, but it is better than someone getting a bad/admin only/debug mob and causing problems.

Also what XDTM said.

**Comment by ma44** [comment]
There's also cameras and dummies (stunnable god mod mobs) that should be blacklisted, it would probably be better to make it anything from the mob/living subtypes then just include any interesting mob/ if you want.

**Comment by factoryman942** [comment]
i can't wait for wizard rounds where the wiz buys poly staff and four apprentices, then spends half an hour in the vacant office converting them all into bubblegums and colossi

**Comment by DaedalusGame** [comment]
rofl this better blacklist megafauna

but also just don't do this at all

**Comment by epochayur** [comment]
>OHHHH BOY PLEASE DO NOT MERGE THIS UNTIL I GET A CHANCE TO TEST IT BECAUSE 9/10 THIS BREAKS SHIT

huh, I wonder why, it's almost as if this is a terrible idea

**Comment by Buggy123** [comment]
Fuck it I'm going to just expand the list a lot anyway so I'll make a new PR

**Comment by ExcessiveUseOfCobblestone** [comment]
@Buggy123 let me clean it up first

**Pull request closed without merging**